activerecord/lib/active_record/associations/nested_error.rb

```diff
@@ -18,13 +18,17 @@ def initialize(association, inner_error)
         def compute_attribute(inner_error)
           association_name = association.reflection.name
 
-          if index_errors_setting && index
+          if collection_association? && index_errors_setting && index
```

Comment:
I think it would be better to introduce `Association#collection?`. It's better than type checking.